src/registration/scripts/directives.js

```diff
@@ -38,9 +50,11 @@
                                     registrationFactory
                                         .register.post(scope.data)
                                         .then(function (response) {
-                                            scope.data = {};
                                             scope.registered = true;
-                                            scope.successData = response.data;
+                                            if (angular.isDefined(scope.onRegister)) {
+                                                scope.onRegister({user: angular.copy(scope.data)});
```

Comment:
Should `scope.data` be `response.data`? They ought to be the same, but maybe the API could return something additional?